core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -0,0 +1,241 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.admin
+
+import java.io.{File, FileInputStream}
+import java.util.Properties
+
+import joptsimple._
+import kafka.security.auth._
+import kafka.server.KafkaConfig
+import kafka.utils._
+import org.apache.kafka.common.security.auth.KafkaPrincipal
+import org.apache.kafka.common.utils.Utils
+
+object AclCommand {
+
+  val delimeter = ","
+  val nl = scala.util.Properties.lineSeparator
+
+  def main(args: Array[String]): Unit = {
+
+    val opts = new AclCommandOptions(args)
+
+    opts.checkArgs()
+
+    val actions = Seq(opts.addOpt, opts.removeOpt, opts.listOpt).count(opts.options.has _)
```

Comment:
Unless you have a specific requirement for a concrete class, it's fine to use the factory methods and let the standard library choose it.